src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,88 @@
+package Duke;
+
+import Commands.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Represents the parser for Duke
+ */
+public class Parser {
+
+    /**
+     * Returns command that would be executed
+     *
+     * @param command Command entered by user
+     * @return Command
+     * @throws DukeException
+     */
+    public static Command parse(String command) throws DukeException {
+        String[] parse = command.split(" ", 2);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM uuuu");
+        switch (parse[0]) {
+        case "bye":
+            checkInputLength(parse, 1);
+            assert parse.length == 1: "Error in input for bye";
+            return new ExitCommand();
+        case "list":
+            checkInputLength(parse, 1);
+            assert parse.length == 1: "Error in input for list";
+            return new ListCommand();
+        case "mark":
+            checkInputLength(parse,2);
```

Comment:
Tiny issue, but note the spacing that is required here: 
```suggestion
            checkInputLength(parse, 2);
```